src/main/java/Parser.java

```diff
@@ -0,0 +1,63 @@
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public enum Parser {
+    BYE("bye"),
+    LIST("list"),
+    TOGGLE("(mark|unmark)\\s(\\d+)"),
+    TODO("todo\\s(.+)"),
+    DEADLINE("deadline\\s(.+)/by\\s(.+)"),
+    EVENT("event\\s(.+)/from\\s(.+)/to\\s(.+)");
+
+    private final String commandRegex;
+
+    Parser(String commandRegex) {
+        this.commandRegex = commandRegex;
+    }
+
+    private String getCommandRegex() {
+        return this.commandRegex;
+    }
+
+    public static Parser analyzeInput(String input) {
+        return Arrays.stream(Parser.values())
+                .filter(token -> input.matches(token.getCommandRegex()))
+                .findFirst()
+                .orElseThrow();
+    }
+
+    public static String[] splitInput(String input) {
+        String[] parsedInput = {};
+
+        Pattern eventPattern = Pattern.compile(EVENT.getCommandRegex());
+        Pattern deadlinePattern = Pattern.compile(DEADLINE.getCommandRegex());
+        Pattern todoPattern = Pattern.compile(TODO.getCommandRegex());
+        Pattern togglePattern = Pattern.compile(TOGGLE.getCommandRegex());
+
+        Matcher eventMatcher = eventPattern.matcher(input);
+        Matcher deadlineMatcher = deadlinePattern.matcher(input);
+        Matcher todoMatcher = todoPattern.matcher(input);
+        Matcher toggleMatcher = togglePattern.matcher(input);
+
```

Comment:
Perhaps having some comments here for explanation will help as it seems like there is quite a bit of things happening